### PR TITLE
Call command / environment with only args they ask for

### PR DESCRIPTION
**Pull request by yuvipanda**
This lets us add more parameters in the future without
breaking backwards compatibility

### CODE REVIEW

File: docs/server-process.rst [hunk 1]
```diff
@@ -60,6 +60,60 @@ pairs.
...
+.. code:: python
+
+   def _cmd_callback(port):
+       return ['some-command', '--port=' + str(port)]
+
+   server_config = {
+       'command': _cmd_callback
+   }
+
+The ``port`` argument will be pased to the callable. This is a simple form of dependency
```
Comment (ryanlovett) [inline docs/server-process.rst]: s/pased/passed/

File: docs/server-process.rst [hunk 2]
```diff
@@ -60,6 +60,60 @@ pairs.
...
+#. **port**
+   The port the command should listen on
+
+If any of the returned strings, lists or dictionaries contain strings
+of form ``{<argument-name>}``, they will be replaced with the value
+of the argument. For example, if your function is:
+
+.. code:: python
+
+   def _openrefine_cmd():
```
Comment (ryanlovett) [inline docs/server-process.rst]: Does this need to be
```suggestion
   def _openrefine_cmd(port):
```

File: jupyter_server_proxy/utils.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+def call_with_asked_args(callback, args):
+    """
+    Call callback only the args it wants from args
```
Comment (ryanlovett) [inline jupyter_server_proxy/utils.py]:
```suggestion
    Call callback with only the args it wants from args
```

File: docs/server-process.rst (hunk 2 above)
Comment (yuvipanda) [reply docs/server-process.rst]: It doesn't! We don't need to use port, so we don't have to ask for it.

**Comment by yuvipanda** [comment]
@ryanlovett thanks for the review! I've fixed them now

File: jupyter_server_proxy/utils.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    # We only pick the local variables
+    asked_arg_names = callback.__code__.co_varnames[:callback.__code__.co_argcount]
+    asked_arg_values = []
+    missing_args = []
+    for asked_arg_name in asked_arg_names:
+        if asked_arg_name in args:
+            asked_arg_values.append(args[asked_arg_name])
+        else:
+            missing_args.append(asked_arg_name)
+    print(asked_arg_values)
```
Comment (manics) [inline jupyter_server_proxy/utils.py]: Is this print statement intentional?

**Comment by yuvipanda** [comment]
@manics I've also invited you to this repo as a collaborator, since I think you are also very interested in seeing this move forward.

**Pull request merged**